src/BaseAmqp.php

```diff
@@ -193,12 +194,12 @@ protected function declareQueue(): void
             $queueOptions->isExclusive(),
             $queueOptions->isAutoDelete(),
             $queueOptions->isNoWait(),
-            $queueOptions->getArguments(),
+            $arguments ? new AMQPTable($arguments) : [],
             $queueOptions->getTicket()
         );
 
         $routingKeys = $queueOptions->getRoutingKeys();
-        if (! \count($routingKeys)) {
+        if (empty($routingKeys)) {
```

Comment:
`$routingKeys` is strictly an array, empty as default. `empty()` function is too much generic and since we should check id `$routingKeys` has elements, `count()` is better.

Much better we should write explicitly:

```php
if (0 === \count($routingKeys)) {
    // ...
}
```

Reply:
You're right, you can use `empty()`.